test/functional/rpc_generateblock.py

```diff
@@ -51,37 +51,35 @@ def run_test(self):
         assert_equal(len(block['tx']), 1)
         assert_equal(block['tx'][0]['vout'][0]['scriptPubKey']['addresses'][0], combo_address)
 
-        # Generate 110 blocks to spend
-        node.generatetoaddress(110, address)
+        # Generate some blocks to spend
+        miniwallet.generate(100)
+        node.generate(100)
 
         # Generate some extra mempool transactions to verify they don't get mined
         for _ in range(10):
-            node.sendtoaddress(address, 0.001)
+            miniwallet.send_self_transfer(from_node=node)
 
         self.log.info('Generate block with txid')
-        txid = node.sendtoaddress(address, 1)
+        txid = miniwallet.send_self_transfer(from_node=node)['txid']
         hash = node.generateblock(address, [txid])['hash']
         block = node.getblock(hash, 1)
         assert_equal(len(block['tx']), 2)
         assert_equal(block['tx'][1], txid)
 
         self.log.info('Generate block with raw tx')
-        utxos = node.listunspent(addresses=[address])
-        raw = node.createrawtransaction([{'txid':utxos[0]['txid'], 'vout':utxos[0]['vout']}],[{address:1}])
-        signed_raw = node.signrawtransactionwithwallet(raw)['hex']
-        hash = node.generateblock(address, [signed_raw])['hash']
+        txid = miniwallet.send_self_transfer(from_node=node)['txid']
```

Comment:
This will add the tx to the mempool, which is not wanted. the raw tx should only be passed to the miner

Reply:
By "not wanted" you mean unnecessary @MarcoFalke? I only ask for my own understanding as the test passes in its current form.

Reply:
Transactions for the miner are picked from the mempool, usually. This test should check that the tx can be supplied in raw form, not taking it from the mempool. Also, it tests that a miner can mine a tx that is not in the mempool.